### ISSUE TITLE
how crazy would this lib be if it supported ranges

### ISSUE THREAD

**Issue by ianstormtaylor** [matthewmueller/date#32]
"i'm available any time **thursday before 2pm**."

would equate to

`Thursday 00:00 – Thursday 14:00`

mind blown. that's all. close this whenever you want ;)

**Comment by gebrits** [comment]
Anyone ever attempted this?

**Comment by kengz** [comment]
yeah that'd be crazy, for example: `last week` would give a pair of start and end dates with a 7-day interval. Like @matthewmueller mentioned, there's 3 types of time-parser usage:

- scheduling (this repo): give a single point in time, future or past
- range: a generalization of scheduling = gives more than one point in time
- cron: gives a cron time pattern, mainly to schedule a cron job.

Of course, apart from these, we need a sufficiently good natural language parsing. This shan't requirement any fancy NLP.

I think imaginable that all these can be unified under a single project?
